zam/webapp/zam_webapp/views/upload_krill.py

```diff
@@ -0,0 +1,160 @@
+import os
+import re
+from tempfile import TemporaryDirectory
+from typing import Iterator, List, Tuple
+
+from pyramid.httpexceptions import HTTPFound
+from pyramid.request import Request
+from pyramid.response import Response
+from pyramid.view import view_config
+from xlrd import open_workbook, XL_CELL_NUMBER
+from xlrd.sheet import Cell
+
+from zam_aspirateur.amendements.models import Amendement
+from zam_aspirateur.amendements.writer import write_json_for_viewer
+
+from zam_visionneuse.loaders import load_aspirateur_source
+from zam_visionneuse.models import load_data
+from zam_visionneuse.templates import render_and_save_html
+
+from zam_webapp.utils import environ
+from zam_webapp.views.lectures import Lecture
+
+
+@view_config(route_name="upload_krill", request_method="POST")
+def upload_krill(request: Request) -> Response:
+
+    next_url = request.route_url(
+        "lecture",
+        chambre=request.matchdict["chambre"],
+        session=request.matchdict["session"],
+        num_texte=request.matchdict["num_texte"],
+    )
+
+    lecture = Lecture(  # type: ignore
+        chambre=request.matchdict["chambre"],
+        session=request.matchdict["session"],
+        num_texte=request.matchdict["num_texte"],
+    )
+
+    # filename = request.POST['xls_file'].filename
+    # log filename?
+
+    xls_file = request.POST["xls_file"].file
+    xls_file.seek(0)
+    xls_data = xls_file.read()
+
+    try:
+        amendements = read_excel_data(xls_data)
+    except BadFormat:
+        request.session.flash(("alert", "Le fichier n'est pas dans le format attendu."))
+        return HTTPFound(location=next_url)
+
+    with TemporaryDirectory() as tmp_dir:
+        filename = os.path.join(tmp_dir, "visionneuse.json")
+
+        write_json_for_viewer(
+            id_projet=0, title=str(lecture), amendements=amendements, filename=filename
+        )
+
+        visionneuse_env = {
+            "ZAM_ASPIRATEUR_SOURCE": filename,
+            "ZAM_JAUNES_SOURCE": tmp_dir,
+        }
+        with environ(visionneuse_env):
+            _, aspirateur_items = load_aspirateur_source()
+            articles, amendements, reponses = load_data([], aspirateur_items)
+            html = render_and_save_html("Titre", articles, amendements, reponses)
+            response = Response(html)
```

Comment:
I'm not sure it should be the responsibility of the repondeur to build that view, to me it should redirect to the visionneuse with a link to the generated json (not yet(?) implemented on the visionneuse side), thoughts?